python_course_1604/class_01/exercize_06_while_break_else.py

```diff
@@ -37,9 +37,12 @@ def count_until(match_value):
     Please use while, break, else construct.
     """
     times = 0
-    while has_next():
-        val = get_next()
-        times += 1
-        if val == match_value:
-            return times
+    while True:
+        if has_next():
+            val = get_next()
+            times += 1
+            if val == match_value:
+                return times
+        else:
```

Comment:
When I wrote exercises I didn’t wrote what I mean. Sorry for that.

During second class, in a slide intended for first class but then moved to second one for reason of time, I explained how to use while statement and its own else. This code would look simpler using it and it would do exactly the same as the else you used with else statement.


On 27 Apr 2016, at 10:58, pamarcos <notifications@github.com<mailto:notifications@github.com>> wrote:


In python_course_1604/class_01/exercize_06_while_break_else.py<https://github.com/FedericoRessi/pythoncourse/pull/16#discussion_r61231976>:

> @@ -37,9 +37,12 @@ def count_until(match_value):
>      Please use while, break, else construct.
>      """
>      times = 0
> -    while has_next():
> -        val = get_next()
> -        times += 1
> -        if val == match_value:
> -            return times
> +    while True:
> +        if has_next():
> +            val = get_next()
> +            times += 1
> +            if val == match_value:
> +                return times
> +        else:


Come on, that's not a break even if it breaks the flow. The instructions say "use while, break, else" :P

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/FedericoRessi/pythoncourse/pull/16/files/9225b4b4d1e43d9ded45e0a3299b9a5618bd5dd3..df155ed592c73759c8a1191fc13325dbbf889a98#r61231976>

--------------------------------------------------------------
Intel Research and Development Ireland Limited
Registered in Ireland
Registered Office: Collinstown Industrial Park, Leixlip, County Kildare
Registered Number: 308263


This e-mail and any attachments may contain confidential material for the sole
use of the intended recipient(s). Any review or distribution by others is
strictly prohibited. If you are not the intended recipient, please contact the
sender and delete all copies.